libs/SmartSync/SmartSync/Classes/Util/SFSyncUpTarget.m

```diff
@@ -123,7 +123,7 @@ - (void)fetchRecordModificationDates:(NSDictionary *)record
     SFRestFailBlock failBlock = ^(NSError *error) {
         [self log:SFLogLevelError format:@"REST request failed with error: %@", error];
         if (modificationResultBlock != NULL) {
-            modificationResultBlock(nil, nil, error);
+            modificationResultBlock(localLastModifiedDate, nil, error);
```

Comment:
Passing nil for both or just the server date has the same result - but it's better to only pass nil for the ones that could not be obtained.